test/api/units_test.rb

```diff
@@ -123,30 +147,67 @@ def assert_tutorial_model_response(response, expected)
     assert_equal response[:abbrev], expected[:abbrev]
   end
 
+  # Test POST for rollover unit should success when missing teaching_period_id
+  def test_rollover_unit_post_success_missing_teaching_period_id
+    unit = Unit.create!(
+      code: "COS10001_TEST",
+      name: "Introduction to Programming",
+      start_date: '2016-05-14T00:00:00.000Z',
+      end_date: '2017-05-14T00:00:00.000Z',
+      description: 'testing'
+    )
+    data_to_post = add_auth_token(
+      start_date: '2016-05-14T00:00:00.000Z',
+      end_date: '2017-05-14T00:00:00.000Z'
+    )
+
+    post_json "/api/units/#{unit.id}/rollover.json", data_to_post
+
+    assert_equal unit.code, last_response_body['code']
+    assert_equal unit.name, last_response_body['name']
+    assert_equal unit.description, last_response_body['description']
+  end
+
+  # Test POST for rollover unit should success when exist teaching_period_id
+  def test_rollover_unit_post_success_exist_teaching_period_id
+    unit = Unit.create!(
+      code: "COS10002_TEST",
+      name: "Introduction to Programming",
+      start_date: '2016-05-14T00:00:00.000Z',
+      end_date: '2017-05-14T00:00:00.000Z',
+      description: 'testing'
+    )
+    data_to_post = add_auth_token(
+      teaching_period_id: 1,
+    )
+
+    post_json "/api/units/#{unit.id}/rollover.json", data_to_post
+
+    assert_equal unit.code, last_response_body['code']
+    assert_equal unit.name, last_response_body['name']
+    assert_equal unit.description, last_response_body['description']
+  end
 
-  def test_addtutorial_to_unit
-    count_tutorials = Tutorial.all.length
+  # Test POST for Upload CSV of all the students in a unit
+  def test_upload_csv_unit_post_success
+    data_to_post = add_auth_token(
+      file: Rack::Test::UploadedFile.new('test_files/COS10001-Outcomes.csv', 'application/csv'),
+    )
 
-    tutorial = {
-      day: 'Wednesday',
-      time: '2:30',
-      location: 'HE12',
-      tutor_username: 'acain',
-      abbrev: 'BC43'
-    }
+    post "/api/csv/units/1", data_to_post
```

Comment:
Please do not use the First Unit to post data into. Please create a new Unit by FactoryBot to post the data...